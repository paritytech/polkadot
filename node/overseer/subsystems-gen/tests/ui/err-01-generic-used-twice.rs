#![allow(dead_code)]

use polkadot_procmacro_overseer_subsystems_gen::AllSubsystemsGen;

#[derive(Clone, AllSubsystemsGen)]
struct AllSubsystems<X> {
	a: X,
	b: X,
}

fn main() {
	let all = AllSubsystems::<u16> {
		a: 0_u16,
		b: 1_u16,
	};
	let _all = all.replace_a(77u8);
}
