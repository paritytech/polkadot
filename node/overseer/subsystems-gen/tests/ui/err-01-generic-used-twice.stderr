error: Generic type parameters may only be used for exactly one field, but is used more than once.
 --> $DIR/err-01-generic-used-twice.rs:7:5
  |
7 |     a: X,
  |        ^

error[E0599]: no method named `replace_a` found for struct `AllSubsystems<u16>` in the current scope
  --> $DIR/err-01-generic-used-twice.rs:16:17
   |
6  | struct AllSubsystems<X> {
   | ----------------------- method `replace_a` not found for this
...
16 |     let _all = all.replace_a(77u8);
   |                    ^^^^^^^^^ method not found in `AllSubsystems<u16>`
